src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDependencyListener.cs

```diff
@@ -138,11 +138,20 @@ internal SqlConnectionContainer(SqlConnectionContainerHashHelper hashHelper, str
                 _appDomainKeyHash = new Dictionary<string, int>(); // Dictionary stores the Start/Stop refcount per AppDomain for this container.
                 _com = new SqlCommand();
                 _com.Connection = _con;
+                bool? dbId = null;
 
                 // SQL BU DT 391534 - determine if broker is enabled on current database.
                 _com.CommandText = "select is_broker_enabled from sys.databases where database_id=db_id()";
 
-                if (!(bool)_com.ExecuteScalar())
+                // db_id() returns the database ID of the current database hence it will always be one line result
+                using SqlDataReader reader = _com.ExecuteReader(CommandBehavior.SingleRow);
+                if (reader.Read() && reader[0] is not null)
+                {
+                    dbId = reader.GetBoolean(0);
+                }
+                reader.Close();
+
+                if (dbId is null || !dbId.Value)
```

Comment:
Can we add test for code that currently throws NRE?

Reply:
This would be shorter as `!value.GetValueOrDefault()`, I'm not sure if it's clearer and perf isn't a major concern just a general one.

Reply:
Regarding the test, it needs more investigation as this only happen on Azure SQL DB and even in this case it may return null.

GetValueOrDefault returns a value even if the HasValue property is false. This also needs more cautious investigation performance wise and also returned value as well.

Reply:
> GetValueOrDefault returns a value even if the HasValue property is false

Yup, and it does so in a single call with a smaller number of instructions than doing the null check then value check. As long as you have that leading negation it would work. But as I said it's hardly a performance problem so leave it as it already is.